lib/vtls/openssl.c

```diff
@@ -3674,6 +3682,8 @@ CURLcode Curl_ssl_setup_x509_store(struct Curl_cfilter *cf,
     }
   }
 
+  ERR_pop_to_mark();
```

Comment:
Not a detailed review, but sure, this matching pair of a set and pop to mark should suffice to avoid later side effects from individual problem CA certs in the store. I hope this will be sufficient to resolve the reported and similar issues.